python/analyze.py

```diff
@@ -5,46 +5,121 @@
 import librosa.display
 import pandas as pd
 from tqdm import tqdm
+import matplotlib.pyplot as plt
+import sklearn
 
 
-def get_mfcc(dir):
+
+def get_features(dir):
+    sr = 22050
+    duration = 5
     try:
-        audio, srate = librosa.load(dir, res_type='kaiser_fast')
-        nffts = [2048, 1024, 512, 256, 128]
-
-        # Ensures that the fft will have adequate frame size
-        for i in nffts:
-            if i <= audio.size:
-                nfft = i
-        nffts = 256
+        audio, srate = librosa.load(dir, sr = sr, mono = True, duration = duration, res_type='kaiser_fast')
+        audio, index = librosa.effects.trim(y = audio, top_db=60, frame_length=2048, hop_length=512)
+        nfft = 2048
+        hop = 1024
+
+        # 0 pad to 2048 minimum size
+        if audio.size < 2048:
+            audio = np.pad(audio, (0, 2048 - audio.size), constant_values=(0, 0))
+
+
         # Max size limits the length of audio sampls so that all features will have the same length (0 padding)
-        max_size = min(88200, audio.size)
-        mfccs = librosa.feature.mfcc(y=audio[:max_size], sr=srate, n_mfcc=40, n_fft=nfft)
-        pad_width = 174 - mfccs.shape[1]
-        mfccs = np.pad(mfccs, pad_width=((0,0), (0, pad_width)), mode='constant')
-        #mfcc = np.mean(mfccs.T, axis=0)
-        return mfccs
+        max_size = min(srate * 3, audio.size)
+        mfccs = librosa.feature.mfcc(y=audio, sr=srate, n_mfcc=40, n_fft=nfft, hop_length = hop, power = 2.0)
```

Comment:
Is `max_size` being used anymore? Shouldn't this be `.mfcc(y=audio[:max_size], ...)` still?

Reply:
is `audio` the entire dir? in that case shouldn't it be something like `audio[:,:max_size]`?